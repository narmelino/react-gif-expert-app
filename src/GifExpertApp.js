import { useState } from "react";
import CategoriesAdd from "./components/CategoriesAdd";
import GifGrid from "./components/GifGrid";

const GifexpertApp = ({ defaultCategories}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <CategoriesAdd setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map(cat => (
          <GifGrid
            key={cat}
            category={cat}
          />))}
      </ol>
    </>
  );
}

GifexpertApp.defaultProps = {
  defaultCategories: [],
}

export default GifexpertApp;