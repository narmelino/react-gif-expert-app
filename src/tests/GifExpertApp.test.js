import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';

describe('Pruebas de <GifExpertApp />', () => {
  test('Debe renderizar el componente correctamente sin lista de categorías.', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe renderizar el componente con una lista de categorías.', () => {
    const categories = ['Goku', 'One Punch']

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});