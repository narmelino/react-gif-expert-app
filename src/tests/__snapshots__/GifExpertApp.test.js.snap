// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp /> Debe renderizar el componente con una lista de categorías. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoriesAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas de <GifExpertApp /> Debe renderizar el componente correctamente sin lista de categorías. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoriesAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
